graphql2/graphqlapp/alert.go

```diff
@@ -39,6 +41,36 @@ func (a *AlertLogEntry) Message(ctx context.Context, obj *alertlog.Entry) (strin
 	return e.String(), nil
 }
 
+func (a *AlertLogEntry) State(ctx context.Context, obj *alertlog.Entry) (*graphql2.AlertLogEntryState, error) {
+	e := *obj
+	meta, ok := e.Meta().(*alertlog.NotificationMetaData)
+	if !ok {
+		return nil, nil
+	}
+
+	s, err := (*App)(a).FindOneAlertLogMessageStatus(ctx, meta.MessageID)
+	if err != nil {
+		return nil, errors.Wrap(err, "find alert log state")
+	}
+
+	status := ""
+	switch s.State {
+	case notification.MessageStateFailedPerm:
```

Comment:
Should we add Temp to this too? Or since it will retry leave it out? It will have an error message until it tries again..

Reply:
Temp works. If it's listed as a "temp" failure we can list it as `Failed. Retrying...` on the UI.